tutorial/20_recipes/009_ask_and_tell.py

```diff
@@ -0,0 +1,237 @@
+"""
+.. _ask_and_tell:
+
+Ask-and-Tell Interface
+=======================
+
+Optuna provides `Ask-and-Tell` interface, a more flexible interface for hyperparameter optimization.
+This tutorial explains three use-cases when the ask-and-tell interface is beneficial:
+
+- :ref:`Apply-optuna-to-an-existing-optimization-problem-with-minimum-modifications`
+- :ref:`Define-and-Run`
+- :ref:`Batch-Optimization`
+
+.. _Apply-optuna-to-an-existing-optimization-problem-with-minimum-modifications:
+
+----------------------------------------------------------------------------
+Apply Optuna to an existing optimization problem with minimum modifications
+----------------------------------------------------------------------------
+
+Let's consider the traditional supervised classification problem; you aim to maximize the validation accuracy.
+To do so, you train `LogisticRegression` as a simple model.
+"""
+import numpy as np
+from sklearn.datasets import make_classification
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import train_test_split
+
+import optuna
+
+
+X, y = make_classification(n_features=10)
+X_train, X_test, y_train, y_test = train_test_split(X, y)
+
+C = 0.01
+clf = LogisticRegression(C=C)
+clf.fit(X_train, y_train)
+val_accuracy = clf.score(X_test, y_test)  # the objective
+
+###################################################################################################
+# Then you try to optimize hyperparameters ``C`` and ``solver`` of the classifier by using optuna.
+# When you introduce optuna naively, you define an ``objective`` function
+# such that it takes ``trial`` and contains ``suggest_*`` functions to sample the hyperparameters:
+
+
+def objective(trial):
+    X, y = make_classification(n_features=10)
+    X_train, X_test, y_train, y_test = train_test_split(X, y)
+
+    C = trial.suggest_loguniform("C", 1e-7, 10.0)
+    solver = trial.suggest_categorical("solver", ("lbfgs", "saga"))
+
+    clf = LogisticRegression(C=C, solver=solver)
+    clf.fit(X_train, y_train)
+    val_accuracy = clf.score(X_test, y_test)
+
+    return val_accuracy
+
+
+study = optuna.create_study(direction="maximize")
+study.optimize(objective, n_trials=10)
+
+###################################################################################################
+# A drawback of this traditional interface is less flexibility.
+# If ``objective`` requires an additional argument for ``objective``,
+# you need to define your customized class for the objective as in
+# `this FAQ section <../../faq.html#how-to-define-objective-functions-that-have-own-arguments>`_.
+# That will make the codebase more complicated.
+# The ask-and-tell interface provides a more flexible syntax to optimize hyperparameters.
+# The following example of ask-and-tell interface is equivalent to the previous code block.
```

Comment:
I do prefer a shorter description, but let @Crissman decide.

The following is my suggestion and not a strong suggestion.

```suggestion
# This interface is not flexible enough.
# For example, if ``objective`` requires additional arguments other than ``trial``,
# you need to define a class as in
# `How to define objective functions that have own arguments? <../../faq.html#how-to-define-objective-functions-that-have-own-arguments>`_.
# The ask-and-tell interface provides a more flexible syntax to optimize hyperparameters.
# The following example is equivalent to the previous code block.
```

Reply:
Sorry but I don't understand your comment enough. Do you mean it's better to leave the first line "A drawback of this traditional interface is less flexibility." as is and replace the other line with my suggestion?